system/Database/Forge.php

```diff
@@ -1001,14 +1006,64 @@ protected function _processPrimaryKeys(string $table): string
         }
 
         if ($this->primaryKeys !== []) {
-            $sql .= ",\n\tCONSTRAINT " . $this->db->escapeIdentifiers('pk_' . $table)
+            if ($asQuery === true) {
+                $sql .= 'ALTER TABLE ' . $this->db->escapeIdentifiers($this->db->DBPrefix . $table) . ' ADD ';
+            } else {
+                $sql .= ",\n\t";
+            }
+            $sql .= 'CONSTRAINT ' . $this->db->escapeIdentifiers('pk_' . $table)
                     . ' PRIMARY KEY(' . implode(', ', $this->db->escapeIdentifiers($this->primaryKeys)) . ')';
         }
 
         return $sql;
     }
 
-    protected function _processIndexes(string $table)
+    /**
```

Comment:
This is the magical method that runs all the indexes just like createTable() does. First we have to load the table fields so the methods work smoothly. Then we process Indexes, then primary keys, and then foreign keys. Some of these methods reset Forge so we have to carry their values and reset them. 
_processIndexes(), _processPrimaryKeys(), and _processForeignKeys() all have the second parameter $asQuery. This retrieves a stand alone query that can be executed outside of the CREATE TABLE statement.